app/src/main/java/agency/five/codebase/android/movieapp/ui/moviedetails/MovieDetailsScreen.kt

```diff
@@ -0,0 +1,228 @@
+package agency.five.codebase.android.movieapp.ui.moviedetails
+
+import agency.five.codebase.android.movieapp.R
+import agency.five.codebase.android.movieapp.mock.MoviesMock
+import agency.five.codebase.android.movieapp.ui.component.*
+import agency.five.codebase.android.movieapp.ui.moviedetails.mapper.MovieDetailsMapper
+import agency.five.codebase.android.movieapp.ui.moviedetails.mapper.MovieDetailsMapperImpl
+import agency.five.codebase.android.movieapp.ui.theme.MovieAppTheme
+import agency.five.codebase.android.movieapp.ui.theme.spacing
+import androidx.compose.foundation.layout.*
+import androidx.compose.foundation.lazy.LazyColumn
+import androidx.compose.foundation.lazy.LazyRow
+import androidx.compose.foundation.lazy.grid.GridCells
+import androidx.compose.foundation.lazy.grid.LazyHorizontalGrid
+import androidx.compose.foundation.lazy.grid.items
+import androidx.compose.foundation.lazy.items
+import androidx.compose.material.MaterialTheme
+import androidx.compose.material.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.runtime.getValue
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.layout.ContentScale
+import androidx.compose.ui.res.colorResource
+import androidx.compose.ui.res.dimensionResource
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.dp
+import androidx.constraintlayout.compose.ConstraintLayout
+import coil.compose.AsyncImage
+
+
+private val movieDetailsMapper: MovieDetailsMapper = MovieDetailsMapperImpl()
+
+val movieDetailsViewState = movieDetailsMapper.toMovieDetailsViewState(MoviesMock.getMovieDetails())
+
+@Composable
+fun MovieDetailsRoute() {
+    val viewState by remember { mutableStateOf(movieDetailsViewState) }
+    MovieDetailsScreen(
+        viewState
+    )
+}
+
+@Composable
+fun MovieDetailsScreen(
+    viewState: MovieDetailsViewState,
+    modifier: Modifier = Modifier
+) {
+    LazyColumn(
+        modifier = modifier
+    ) {
+        item {
+            MovieDetailsTitle(viewState) { viewState.isFavorite = !viewState.isFavorite }
+        }
+        item {
+            MovieDetailsOverView(
+                viewState,
+                Modifier.padding(horizontal = MaterialTheme.spacing.medium)
+            )
+        }
+        item {
+            Spacer(modifier = Modifier.height(MaterialTheme.spacing.small))
+            MovieDetailsTopCast(viewState)
+        }
+    }
+}
+
+@Composable
+fun MovieDetailsTitle(
+    viewState: MovieDetailsViewState,
+    onLikeButtonClick: () -> Unit,
+) {
+    ConstraintLayout(
```

Comment:
this can be oneliner

Reply:
Also, is it necessary for this to be ConstraintLayout?

ConstraintLayout has Image linked to `parent.top` and Column linked to `image.bottom` - you can use Column instead